test/irb/yamatanooroti/test_rendering.rb

```diff
@@ -311,154 +282,129 @@ def test_autocomplete_with_showdoc_in_gaps_on_narrow_screen_left
     write_irbrc <<~'LINES'
-      puts 'start IRB'
       require "irb/pager"
     LINES
-    start_terminal(10, 80, %W{ruby -I#{@pwd}/lib #{@pwd}/exe/irb}, startup_message: 'start IRB')
+    start_terminal(10, 80, %W{ruby -I#{@pwd}/lib #{@pwd}/exe/irb}, startup_message: /irb\(main\)/)
     write("IRB::Pager.page_content('a' * (80 * 8))\n")
     write("'foo' + 'bar'\n") # eval something to make sure IRB resumes
-    close
 
-    screen = result.join("\n").sub(/\n*\z/, "\n")
-    assert_match(/a{80}/, screen)
+    assert_screen(/a{80}/)
     # because pager is invoked, foobar will not be evaluated
-    assert_not_match(/foobar/, screen)
+    assert_screen(/\A(?!foobar)/)
+    close
   end
 
   def test_pager_page_content_pages_output_when_it_does_not_fit_in_the_screen_because_of_screen_height
     write_irbrc <<~'LINES'
-      puts 'start IRB'
       require "irb/pager"
     LINES
-    start_terminal(10, 80, %W{ruby -I#{@pwd}/lib #{@pwd}/exe/irb}, startup_message: 'start IRB')
+    start_terminal(10, 80, %W{ruby -I#{@pwd}/lib #{@pwd}/exe/irb}, startup_message: /irb\(main\)/)
     write("IRB::Pager.page_content('a\n' * 8)\n")
     write("'foo' + 'bar'\n") # eval something to make sure IRB resumes
-    close
 
-    screen = result.join("\n").sub(/\n*\z/, "\n")
-    assert_match(/(a\n){8}/, screen)
+    assert_screen(/(a\n){8}/)
     # because pager is invoked, foobar will not be evaluated
-    assert_not_match(/foobar/, screen)
+    assert_screen(/\A(?!foobar)/)
+    close
   end
 
   def test_pager_page_content_doesnt_page_output_when_it_fits_in_the_screen
     write_irbrc <<~'LINES'
-      puts 'start IRB'
       require "irb/pager"
     LINES
-    start_terminal(10, 80, %W{ruby -I#{@pwd}/lib #{@pwd}/exe/irb}, startup_message: 'start IRB')
+    start_terminal(10, 80, %W{ruby -I#{@pwd}/lib #{@pwd}/exe/irb}, startup_message: /irb\(main\)/)
     write("IRB::Pager.page_content('a' * (80 * 7))\n")
     write("'foo' + 'bar'\n") # eval something to make sure IRB resumes
-    close
 
-    screen = result.join("\n").sub(/\n*\z/, "\n")
-    assert_match(/a{80}/, screen)
+    assert_screen(/a{80}/)
     # because pager is not invoked, foobar will be evaluated
-    assert_match(/foobar/, screen)
+    assert_screen(/foobar/)
+    close
   end
 
   def test_long_evaluation_output_is_paged
     write_irbrc <<~'LINES'
-      puts 'start IRB'
       require "irb/pager"
     LINES
-    start_terminal(10, 80, %W{ruby -I#{@pwd}/lib #{@pwd}/exe/irb}, startup_message: 'start IRB')
+    start_terminal(10, 80, %W{ruby -I#{@pwd}/lib #{@pwd}/exe/irb}, startup_message: /irb\(main\)/)
     write("'a' * 80 * 11\n")
     write("'foo' + 'bar'\n") # eval something to make sure IRB resumes
-    close
 
-    screen = result.join("\n").sub(/\n*\z/, "\n")
-    assert_match(/(a{80}\n){8}/, screen)
+    assert_screen(/(a{80}\n){8}/)
     # because pager is invoked, foobar will not be evaluated
-    assert_not_match(/foobar/, screen)
+    assert_screen(/\A(?!foobar)/)
```

Comment:
Yamatanooroti does not have `assert_not_screen`. We need to use negative lookahead to use a retryable assertion.